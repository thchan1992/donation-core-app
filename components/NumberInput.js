import { StyleSheet, Text, TextInput, View } from "react-native";

const NumberInput = ({
  value,
  setter,
  placeholder,
  height,
  maxWidth,
  minWidth,
  maxLength,
  isPound,
}) => {
  return (
    <View style={{ flexDirection: "row" }}>
      {isPound != null && <Text style={styles.poundsSign}>Â£</Text>}
      <TextInput
        textAlign={"center"}
        placeholderTextColor={"grey"}
        onChangeText={setter}
        maxLength={maxLength}
        style={[
          styles.input,
          {
            minheight: height,
            fontSize: height,
            maxWidth: maxWidth,
            minWidth: minWidth,
          },
        ]}
        value={value}
        placeholder={placeholder}
        keyboardType="numeric"
      />
    </View>
  );
};

export default NumberInput;

const styles = StyleSheet.create({
  input: {
    margin: 12,
    borderWidth: 1,
    padding: 10,
    borderColor: "#FF304F",
    borderRadius: 8,
    minheight: 50,
    fontSize: 50,
    color: "white",
    fontSize: 50,
    maxWidth: 200,
  },
  poundsSign: {
    fontSize: 50,
    color: "white",
    paddingTop: 20,
  },
});
