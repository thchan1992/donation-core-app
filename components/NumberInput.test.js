import renderer from "react-test-renderer";
import React from "react";
import NumberInput from "./NumberInput";

describe("NumberInput", () => {
  let setter;

  beforeEach(() => {
    setter = jest.fn();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <NumberInput
          value="123"
          setter={setter}
          placeholder="Test Placeholder"
          height={50}
          maxWidth={200}
          minWidth={150}
          maxLength={5}
          isPound={true}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls setter when input value changes", () => {
    const component = renderer.create(
      <NumberInput
        value="123"
        setter={setter}
        placeholder="Test Placeholder"
        height={50}
        maxWidth={200}
        minWidth={150}
        maxLength={5}
        isPound={true}
      />
    );
    const instance = component.root;
    const input = instance.findByType("TextInput");
    input.props.onChangeText("456");
    expect(setter).toHaveBeenCalledWith("456");
  });
});
