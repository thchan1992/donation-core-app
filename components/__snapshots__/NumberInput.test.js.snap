// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 50,
        "paddingTop": 20,
      }
    }
  >
    Â£
  </Text>
  <TextInput
    keyboardType="numeric"
    maxLength={5}
    onChangeText={[MockFunction]}
    placeholder="Test Placeholder"
    placeholderTextColor="grey"
    style={
      Array [
        Object {
          "borderColor": "#FF304F",
          "borderRadius": 8,
          "borderWidth": 1,
          "color": "white",
          "fontSize": 50,
          "margin": 12,
          "maxWidth": 200,
          "minheight": 50,
          "padding": 10,
        },
        Object {
          "fontSize": 50,
          "maxWidth": 200,
          "minWidth": 150,
          "minheight": 50,
        },
      ]
    }
    textAlign="center"
    value="123"
  />
</View>
`;
